src/packagedcode/pypi.py

```diff
@@ -272,9 +274,9 @@ def parse_with_dparse(location):
     return package_dependencies
 
 
-def parse_requirements_txt(location):
+def parse_dependency_file(location):
```

Comment:
We should probably rename this `parse_python_dependency_file()`, as this is what it actually does.

Reply:
We need to create different functions for all the dependency file. While parsing `requirements.txt` we don't need to parse the file again to detect extra data and parsing `Pipfile.lock` we need to parse the file again to extract `sha256` because `dparse` does not contain code to extract `sha256` from `_meta`.